Maple2.File.IO/Nif/PhysXMesh.cs

```diff
@@ -0,0 +1,172 @@
+﻿using Maple2.File.IO.Enum;
+using System.Numerics;
+using System.Text;
+
+namespace Maple2.File.IO.Nif;
+
+public struct PhysXMeshFace {
+    public byte Vert0;
+    public byte Vert1;
+    public byte Vert2;
+
+    public PhysXMeshFace(byte vert0 = 0, byte vert1 = 0, byte vert2 = 0) {
+        Vert0 = vert0;
+        Vert1 = vert1;
+        Vert2 = vert2;
+    }
+}
+
+public class PhysXMesh {
+    public NxsMeshType Type { get; init; }
+    public List<Vector3> Vertices { get; init; }
+    public List<PhysXMeshFace> Faces { get; init; }
+
+    public PhysXMesh(byte[] data) {
+        Vertices = new List<Vector3>();
+        Faces = new List<PhysXMeshFace>();
+
+        Type = ParseNxsMesh(data);
+    }
+
+    private NxsMeshType ParseNxsMesh(byte[] data) {
+        if (data.Length < 8) {
+            throw new InvalidDataException("PhysX mesh data too small to fit header");
+        }
+
+        string headerPiece1 = Encoding.UTF8.GetString(data, 0, 3);
+        string headerPiece2 = Encoding.UTF8.GetString(data, 4, 4);
+
+        if (headerPiece1 != "NXS") {
+            throw new InvalidDataException($"Invalid header found in PhysX mesh data: {Encoding.UTF8.GetString(data, 0, 8)}");
+        }
+
+        switch (headerPiece2) {
+            case "CVXM":
+                return ParseConvexMesh(data);
+            case "MESH":
+                return ParseTriangleMesh(data);
+            case "CLTH":
+                throw new NotSupportedException($"Cloth mesh not supported! Found unsupported PhysX cloth mesh in mesh data");
+            default:
+                throw new InvalidDataException($"Unknown PhysX nxs mesh type {headerPiece2} found in mesh data");
+        }
+    }
+
+    private NxsMeshType ParseConvexMesh(byte[] data) {
+        EndianReader reader = new EndianReader(data, !BitConverter.IsLittleEndian, 8);
+
+        uint unk1 = reader.ReadUInt32();
+        uint unk2 = reader.ReadUInt32();
+
+        bool unkDataSectionMatches = false;
+
+        unkDataSectionMatches |= data[reader.Index + 0] == 'I';
```

Comment:
what even is this? i feel like you should write it as hex and not random characters

Also why +0? seems redundant

Reply:
- There's some data tags in some kind & I wanted to do a check to see if they're always formatted that way as a sort of best guess measure of detecting when the parser would fail on the nxs mesh data. It's remained the same for all of the NIFs in GMS2. I haven't found the format specifications yet, because most people just feed it in to the PhysX API where their parser is hidden & buried. The reason for this parser existing is for extracting the meshes for use in navmesh generation.

Reply:
Also I just noticed I forgot to put the real offsets...